sdk/documentintelligence/ai-document-intelligence-rest/test/public/analysis.spec.ts

```diff
@@ -0,0 +1,883 @@
+
+      const base64Source = fs.readFileSync(filePath, { encoding: "base64" });
+
+      const initialResponse = await client
+        .path("/documentModels/{modelId}:analyze", "prebuilt-receipt")
+        .post({
+          contentType: "application/json",
+          body: {
+            base64Source,
+          },
+        });
+
+      if (isUnexpected(initialResponse)) {
+        throw initialResponse.body.error;
+      }
+
+      const poller = getLongRunningPoller(client, initialResponse, { ...testPollingOptions });
+      const analyzeResult = (
+        (await (await poller).pollUntilDone()).body as AnalyzeResultOperationOutput
+      ).analyzeResult;
+      const documents = analyzeResult?.documents;
+      assert.isNotEmpty(documents);
+
+      assert.equal(documents?.[0].docType, "receipt.retailMeal");
+    });
+
+    it("url", async () => {
+      const url = makeTestUrl("/contoso-allinone.jpg");
+
+      const initialResponse = await client
+        .path("/documentModels/{modelId}:analyze", "prebuilt-receipt")
+        .post({
+          contentType: "application/json",
+          body: {
+            urlSource: url,
+          },
+        });
+
+      if (isUnexpected(initialResponse)) {
+        throw initialResponse.body.error;
+      }
+
+      const poller = getLongRunningPoller(client, initialResponse, { ...testPollingOptions });
+      const analyzeResult = (
+        (await (await poller).pollUntilDone()).body as AnalyzeResultOperationOutput
+      ).analyzeResult;
+      const documents = analyzeResult?.documents;
+      assert.isNotEmpty(documents);
+
+      assert.equal(documents?.[0].docType, "receipt.retailMeal");
+    });
+
+    it("specifying locale", async () => {
+      const url = makeTestUrl("/contoso-allinone.jpg");
+
+      const initialResponse = await client
+        .path("/documentModels/{modelId}:analyze", "prebuilt-receipt")
+        .post({
+          contentType: "application/json",
+          body: {
+            urlSource: url,
+          },
+          queryParameters: { locale: "en-IN" },
+        });
+
+      if (isUnexpected(initialResponse)) {
+        throw initialResponse.body.error;
+      }
+
+      const poller = getLongRunningPoller(client, initialResponse, { ...testPollingOptions });
+      (await (await poller).pollUntilDone()).body as AnalyzeResultOperationOutput;
+    });
+
+    it("invalid locale throws", async () => {
+      const url = makeTestUrl("/contoso-allinone.jpg");
+
+      try {
+        await client.path("/documentModels/{modelId}:analyze", "prebuilt-receipt").post({
+          contentType: "application/json",
+          body: {
+            urlSource: url,
+          },
+          queryParameters: { locale: "thisIsNotAValidLanguage" },
+        });
+
+        assert.fail("Expected an exception due to invalid locale.");
+      } catch (ex: any) {
+        // Just make sure we didn't get a bad error message
+        assert.isFalse((ex as Error).message.includes("<empty>"));
```

Comment:
Wouldn't the catch block also swallows the exception from `assert.fail("Expected an exception due to invalid locale.")`?

Reply:
Oops, overlooked the copilot generated catch block

Reply:
fixed